test/e2e/tests/test_user_group.py

```diff
@@ -0,0 +1,96 @@
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the Elasticache UserGroup resource
+"""
+import boto3
+import pytest
+import logging
+
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_elasticache_resource
+
+RESOURCE_PLURAL = "usergroups"
+
+
+# factory for user_group
+@pytest.fixture(scope="module")
+def make_user_group():
+    def _make_user_group(yaml_name, input_dict, user_group_name):
+        usergroup = load_elasticache_resource(
+            yaml_name, additional_replacements=input_dict)
+        logging.debug(usergroup)
+
+        reference = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL, user_group_name, namespace="default")
+        _ = k8s.create_custom_resource(reference, usergroup)
+        resource = k8s.wait_resource_consumed_by_controller(reference, wait_periods=10)
+        assert resource is not None
+        return (reference, resource)
+
+    return _make_user_group
+
+@pytest.fixture(scope="module")
+def make_non_default_user():
+    ec = boto3.client("elasticache")
+    user_id = random_suffix_name("ackecuser", 32)
+
+    _ = ec.create_user(UserId=user_id,
+                       UserName="ACKNonDefaultUser",
+                       Engine="Redis",
+                       NoPasswordRequired=True,
+                       AccessString="on -@all")
+
+    yield user_id
+
+    ec.delete_user(UserId=user_id)
+
+# setup/teardown for test_user_group_create_update
+@pytest.fixture(scope="module")
+def user_group_create(make_user_group):
+    user_group_name = random_suffix_name("ack-usergroup", 32)
+
+    input_dict = {
+        "USER_GROUP_ID": user_group_name,
+    }
+
+    # Create new user group
+    (reference,resource) = make_user_group("usergroup_default", input_dict, input_dict['USER_GROUP_ID'])
+
+    yield (reference, resource)
+
+    # Teardown
+    k8s.delete_custom_resource(reference)
+
+
+@service_marker
+class TestUserGroup:
+    def test_user_group_create_update(self, user_group_create, make_non_default_user):
+        (reference, _) = user_group_create
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=15)
+
+        update_dict = {
+            "USER_GROUP_ID": reference.name,
+            "USER": make_non_default_user
+        }
+
+        # Update the user group
+        updated_spec = load_elasticache_resource(
+            "usergroup_additional_user", additional_replacements=update_dict)
+
+        k8s.patch_custom_resource(reference, updated_spec)
```

Comment:
I think you may be able to remove the call to `load_elasticache_resource` by doing something like the below:

```
patch = {"spec": {"userIDs": ['default', make_non_default_user]}}
k8s.patch_custom_resource(reference, patch)
```

or just combine those into one line